solr/core/src/java/org/apache/solr/cluster/events/impl/DelegatingClusterEventProducer.java

```diff
@@ -90,7 +96,9 @@ public void setDelegate(ClusterEventProducer newDelegate) {
         log.debug("--- delegate {} already in state {}", delegate, delegate.getState());
       }
     }
-    this.version++;
+    if (versionTracker != null) {
```

Comment:
We have a synchronization issue (memory barrier type, not concurrent access type). The thread calling `setDelegate()` is accessing `versionTracker` set by another thread without synchronization.
Can be fixed by making `versionTracker` volatile.